tools/src/test/java/org/apache/kafka/tools/TopicCommandIntegrationTest.java

```diff
@@ -787,16 +827,28 @@ public void testDescribeUnderReplicatedPartitionsWhenReassignmentIsInProgress(St
 
         // describe the topic and test if it's under-replicated
         String simpleDescribeOutput = captureDescribeTopicStandardOut(buildTopicCommandOptionsWithBootstrap("--describe", "--topic", testTopicName));
-        String[] simpleDescribeOutputRows = simpleDescribeOutput.split("\n");
-        assertTrue(simpleDescribeOutputRows[0].startsWith(String.format("Topic: %s", testTopicName)));
-        assertEquals(2, simpleDescribeOutputRows.length);
+        String[] simpleDescribeOutputRows = simpleDescribeOutput.split(System.lineSeparator());
+        assertTrue(simpleDescribeOutputRows[0].startsWith(String.format("Topic: %s", testTopicName)),
+                "Unexpected describe output: " + simpleDescribeOutputRows[0]);
+        assertEquals(2, simpleDescribeOutputRows.length,
+                "Unexpected describe output length: " + simpleDescribeOutputRows.length);
 
         String underReplicatedOutput = captureDescribeTopicStandardOut(buildTopicCommandOptionsWithBootstrap("--describe", "--under-replicated-partitions"));
         assertEquals("", underReplicatedOutput,
             String.format("--under-replicated-partitions shouldn't return anything: '%s'", underReplicatedOutput));
 
         // Verify reassignment is still ongoing.
-        PartitionReassignment reassignments = adminClient.listPartitionReassignments(Collections.singleton(tp)).reassignments().get().get(tp);
+        PartitionReassignment reassignments = null;
+        int retryCount = 0;
+        int maxRetries = 20;
+        while (reassignments == null && retryCount < maxRetries) {
+            reassignments = adminClient.listPartitionReassignments(Collections.singleton(tp)).reassignments().get().get(tp);
+            if (reassignments != null) {
+                break;
+            }
+            retryCount++;
+            Thread.sleep(100L);
+        }
         assertFalse(reassignments.addingReplicas().isEmpty());
```

Comment:
Could you explain why we need this change? And why can't we use `TestUtils.waitUntilTrue`?

Reply:
Thanks. I've used `waitUntilTrue` to implement the retry logic.